src/experimental/constructs/iam/roles/cross-account-role.test.ts

```diff
@@ -0,0 +1,62 @@
+import { Template } from "aws-cdk-lib/assertions";
+import { ServicePrincipal } from "aws-cdk-lib/aws-iam";
+import { GuRole } from "../../../../constructs/iam";
+import { simpleGuStackForTesting } from "../../../../utils/test";
+import { GuCrossAccountRoleExperimental } from "./cross-account-role";
+
+describe("The GuCrossAccountRoleExperimental construct", () => {
+  it("can create a cross account role", () => {
+    const stack = simpleGuStackForTesting();
+    new GuCrossAccountRoleExperimental(stack, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "nameOfRoleInOtherAccountWhichCanAssumeThis-STAGE",
+      roleName: "crossAccountRole",
+      accountId: "1234",
+    });
+    expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
+  });
+
+  it("can create a cross account role that can be assumed by a service in another account", () => {
+    const stackThatCreatesTheRole = simpleGuStackForTesting();
+    new GuCrossAccountRoleExperimental(stackThatCreatesTheRole, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "nameOfRoleInOtherAccountWhichCanAssumeThisNewlyCreatedOne-CODE",
+      roleName: "crossAccountRole",
+      accountId: "1234",
+    });
+
```

Comment:
Minor: it might be worth adding an example policy to `GuCrossAccountRoleExperimental` just to give people a simple demo of how this can be done (I'm assuming that they might copy & paste this test code!)